aspnetcore/grpc/interprocess-uds.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Inter-process communication with gRPC and Unix domain sockets
+author: jamesnk
+description: Learn how to use gRPC for inter-process communication with Unix domain sockets
+monikerRange: '>= aspnetcore-5.0'
+ms.author: jamesnk
+ms.date: 08/08/2022
+uid: grpc/interprocess-uds
+---
+# Inter-process communication with gRPC and Unix domain sockets
+
+By [James Newton-King](https://twitter.com/jamesnk)
+
+.NET supports inter-process communication (IPC) using gRPC. For more information about getting started with using gRPC to communicate between processes, see [Inter-process communication with gRPC](xref:grpc/interprocess).
+
+[Unix domain sockets (UDS)](https://wikipedia.org/wiki/Unix_domain_socket) is a widely supported IPC transport that's more efficient than TCP when the client and server are on the same machine. This article discusses how to configure gRPC communication over UDS.
+
+Requirements:
+
+* .NET 5 or later
+* Linux, macOS, or [Windows 10/Windows Server 2019 or later](https://devblogs.microsoft.com/commandline/af_unix-comes-to-windows/)
+
+## Server configuration
+
+Unix domain sockets are supported by [Kestrel](xref:fundamentals/servers/kestrel), which is configured in `Program.cs`:
+
+```csharp
+public static readonly string SocketPath = Path.Combine(Path.GetTempPath(), "socket.tmp");
+
+public static IHostBuilder CreateHostBuilder(string[] args) =>
+    Host.CreateDefaultBuilder(args)
+        .ConfigureWebHostDefaults(webBuilder =>
+        {
+            webBuilder.UseStartup<Startup>();
+            webBuilder.ConfigureKestrel(options =>
+            {
+                if (File.Exists(SocketPath))
+                {
+                    File.Delete(SocketPath);
+                }
+                options.ListenUnixSocket(SocketPath, listenOptions =>
+                {
+                    listenOptions.Protocols = HttpProtocols.Http2;
+                });
+            });
+        });
+```
+
+The preceding example:
+
+* Configures Kestrel's endpoints in `ConfigureKestrel`.
+* Calls <xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions.ListenUnixSocket%2A> to listen to a UDS with the specified path.
+* Creates a UDS endpoint that isn't configured to use HTTPS. For information about enabling HTTPS, see [Kestrel HTTPS endpoint configuration](xref:fundamentals/servers/kestrel/endpoints#listenoptionsusehttps).
+
+### Client configuration
+
+`GrpcChannel` supports making gRPC calls over custom transports. When a channel is created, it can be configured with a `SocketsHttpHandler` that has a custom `ConnectCallback`. The callback allows the client to make connections over custom transports and then send HTTP requests over that transport.
```

Comment:
```suggestion
`GrpcChannel` supports making gRPC calls over custom transports. When a channel is created, it can be configured with a <xref:System.Net.Http.SocketsHttpHandler> that has a custom <xref:System.Net.Http.SocketsHttpHandler.ConnectCallback>. The callback allows the client to make connections over custom transports and then send HTTP requests over that transport.
```